cloudferrylib/os/storage/cinder_storage.py

```diff
@@ -462,3 +542,283 @@ def get_volume_path_iscsi(self, vol_id):
             lun)
 
         return volume_path
+
+
+def skip_invalid_status_volumes(volumes):
+    result = []
+    for v in volumes:
+        if v[STATUS] not in VALID_STATUSES:
+            LOG.warning('Skipping volume %s[%s] in "%s" state',
+                        v.get(DISPLAY_NAME, ''), v[ID], v[STATUS])
+        else:
+            result.append(v)
+    return result
+
+
+class CinderNFSStorage(CinderStorage):
```

Comment:
`CinderNFSStorage` class and `skip_invalid_status_volumes` function were removed earlier, which means your code is rebased incorrectly, please update.